### ISSUE TITLE
Overrides has a hard error when string type hints fail to evaluate

### ISSUE THREAD

**Issue by dpcollins-google** [mkorpela/overrides#79]
For the following class:

```
from concurrent.futures import Future

class MyInterface(ABC):
  @abstractmethod
  def run(self) -> "Future[str]":
    pass
```

Overrides will fail with a hard error due to:

```
  File "lib/python3.6/site-packages/overrides/overrides.py", line 67, in overrides
    return _overrides(method, check_signature, check_at_runtime)
  File "lib/python3.6/site-packages/overrides/overrides.py", line 93, in _overrides
    _validate_method(method, super_class, check_signature)
  File "lib/python3.6/site-packages/overrides/overrides.py", line 114, in _validate_method
    ensure_signature_is_compatible(super_method, method, is_static)
  File "lib/python3.6/site-packages/overrides/signature.py", line 75, in ensure_signature_is_compatible
    super_type_hints = _get_type_hints(super_callable)
  File "lib/python3.6/site-packages/overrides/signature.py", line 46, in _get_type_hints
    return get_type_hints(callable)
  File "lib/python3.6/typing.py", line 1543, in get_type_hints
    value = _eval_type(value, globalns, localns)
  File "lib/python3.6/typing.py", line 350, in _eval_type
    return t._eval_type(globalns, localns)
  File "lib/python3.6/typing.py", line 245, in _eval_type
    eval(self.__forward_code__, globalns, localns),
  File "<string>", line 1, in <module>
TypeError: 'type' object is not subscriptable
```

Hard failure may not be the correct answer here: IMO the unevaluatable type hint should be ignored.

**Comment by mkorpela** [comment]
Thank you for the report.

**Comment by dpcollins-google** [comment]
Addendum following the response on the underlying library. I know this is incorrect. However, there is no way to express "concurrent.futures.Future (not asyncio.Future) with a given return type" in python < 3.9. The fact that the type hint is part of a string means it does not need to be possible to evaluate it. My suggestion is instead, if the type hint is a string type hint, and the library fails to determine what type it is, to be permissive and allow it anyway. I don't know how hard this would be to do.

**Comment by mkorpela** [comment]
Just verified that everything works fine with Python 3.9.

**Comment by mkorpela** [comment]
Fixed in 6.0.1. https://pypi.org/project/overrides/6.0.1/

**Comment by dpcollins-google** [comment]
Thanks!
